pkgs/by-name/na/nautilus/package.nix

```diff
@@ -35,105 +37,121 @@
+      gsettings-desktop-schemas
+      gnome-user-share
+      gst_all_1.gst-plugins-base
+      gtk4
+      libadwaita
+      libportal-gtk4
+      libexif
+      libnotify
+      libseccomp
+      libselinux
+      gdk-pixbuf
+      libcloudproviders
+      shared-mime-info
+      tracker
+      tracker-miners
+      gnome-autoar
+    ];
 
-  buildInputs = [
-    gexiv2
-    glib-networking
-    gnome-desktop
-    gnome.adwaita-icon-theme
-    gsettings-desktop-schemas
-    gnome-user-share
-    gst_all_1.gst-plugins-base
-    gtk4
-    libadwaita
-    libportal-gtk4
-    libexif
-    libnotify
-    libseccomp
-    libselinux
-    gdk-pixbuf
-    libcloudproviders
-    shared-mime-info
-    tracker
-    tracker-miners
-    gnome-autoar
-  ];
+    propagatedBuildInputs = [
+      gtk4
+    ];
 
-  propagatedBuildInputs = [
-    gtk4
-  ];
+    mesonFlags = [
+      "-Ddocs=true"
+    ];
 
-  mesonFlags = [
-    "-Ddocs=true"
-  ];
+    preFixup = ''
+      gappsWrapperArgs+=(
+        # Thumbnailers
+        --prefix XDG_DATA_DIRS : "${gdk-pixbuf}/share"
+        --prefix XDG_DATA_DIRS : "${libjxl}/share"
+        --prefix XDG_DATA_DIRS : "${librsvg}/share"
+        --prefix XDG_DATA_DIRS : "${webp-pixbuf-loader}/share"
+        --prefix XDG_DATA_DIRS : "${shared-mime-info}/share"
+      )
+    '';
 
-  preFixup = ''
-    gappsWrapperArgs+=(
-      # Thumbnailers
-      --prefix XDG_DATA_DIRS : "${gdk-pixbuf}/share"
-      --prefix XDG_DATA_DIRS : "${libjxl}/share"
-      --prefix XDG_DATA_DIRS : "${librsvg}/share"
-      --prefix XDG_DATA_DIRS : "${webp-pixbuf-loader}/share"
-      --prefix XDG_DATA_DIRS : "${shared-mime-info}/share"
-    )
-  '';
+    postFixup = ''
+      # Cannot be in postInstall, otherwise _multioutDocs hook in preFixup will move right back.
+      moveToOutput "share/doc" "$devdoc"
+    '';
 
-  postFixup = ''
-    # Cannot be in postInstall, otherwise _multioutDocs hook in preFixup will move right back.
-    moveToOutput "share/doc" "$devdoc"
-  '';
+    passthru = {
+      updateScript = gnome.updateScript {
+        packageName = "nautilus";
+      };
+    };
 
-  passthru = {
-    updateScript = gnome.updateScript {
-      packageName = "nautilus";
+    meta = with lib; {
+      description = "File manager for GNOME";
+      homepage = "https://apps.gnome.org/Nautilus/";
+      license = licenses.gpl3Plus;
+      platforms = platforms.linux;
+      maintainers = teams.gnome.members;
+      mainProgram = "nautilus";
     };
-  };
+  });
 
-  meta = with lib; {
-    description = "File manager for GNOME";
-    homepage = "https://apps.gnome.org/Nautilus/";
-    license = licenses.gpl3Plus;
-    platforms = platforms.linux;
-    maintainers = teams.gnome.members;
-    mainProgram = "nautilus";
-  };
-})
+in
+if nautilusExtensions == [ ] then
+  unwrapped
+else
+  import ./wrapper.nix {
+    inherit
+      makeWrapper
+      symlinkJoin
+      nautilusExtensions
+      lib
+      ;
+    nautilus = unwrapped;
+  }
```

Comment:
Maybe we can put this logic in the service, and expose in `pkgs` both `nautilus-with-extensions` and `nautilus`? This way also the diff will be smaller.

Reply:
I can imagine having something like `makeExtensionWrapper { varName = "NAUTILUS_4_EXTENSION_DIR"; extensionPath = "lib/…"; package = unwrapped; plugins = nautilusExtensions; }` that would scan the whole derivation but there are several design questions that I did not want to dig into now. We can introduce such helper function later.

I did not want to have two different attributes since that is confusing for some users (based on questions I have seen), see also the commit message.

Reply:
> I can imagine having something like `makeExtensionWrapper { varName = "NAUTILUS_4_EXTENSION_DIR"; extensionPath = "lib/…"; package = unwrapped; plugins = nautilusExtensions; }` that would scan the whole derivation but there are several design questions that I did not want to dig into now. We can introduce such helper function later.
> 
> I did not want to have two different attributes since that is confusing for some users (based on questions I have seen), see also the commit message.

Fair enough. Especially when the attribute eventually evaluates to the unwrapped derivation if there are no extensions enabled.